uftrace.c

```diff
@@ -815,15 +823,12 @@ static error_t parse_option(int key, char *arg, struct argp_state *state)
 			argp_usage(state);
 
 		if (opts->exename == NULL) {
-			switch (opts->mode) {
-			case UFTRACE_MODE_RECORD:
-			case UFTRACE_MODE_LIVE:
+			if (opts->mode == UFTRACE_MODE_LIVE
+			    || (opts->mode == UFTRACE_MODE_RECORD
+				&& opts->pid <= 0)) {
```

Comment:
The operator should be put on the above line like `opts->mode == UFTRACE_MODE_RECORD &&`.  Also why not keep switch statement and just adding the pid check?